x/auth/internal/context/context.go

```diff
@@ -0,0 +1,141 @@
+package context
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+type AuthContext interface {
```

Comment:
Woaaaa I do not think this is the pattern we want here. Why remove this from the client context?

Reply:
This PR should be closed really - it's basically my sketch pad. We need to generalise `auth` basic types first

Reply:
Cool cool 👍

Reply:
this looks like an expected keeper almost